modules/StreamingAPI/src/main/java/org/gephi/streaming/api/GraphUpdaterEventHandler.java

```diff
@@ -0,0 +1,563 @@
+            /*
+             * Unsupported set of SOURCE and TARGET
+             */
+//            case SOURCE:
+//                String source = value.toString();
+//                if (source != null) {
+//                    Node sourceNode = graph.getNode(source);
+//                    edge.setSource(sourceNode);
+//                }
+//                break;
+//            case TARGET:
+//                String target = value.toString();
+//                if (target != null) {
+//                    Node targetNode = graph.getNode(target);
+//                    edge.setTarget(targetNode);
+//                }
+//                break;
+            case WEIGHT:
+                float weight = Float.valueOf(value.toString());
+                if (weight != 0) {
+                    edge.setWeight(weight);
+                }
+                break;
+            case R:
+                float r = Float.valueOf(value.toString());
+                if (r < 0.0 || r > 1.0) {
+                    throw new IllegalArgumentException("Color parameter outside of expected range: " + r);
+                }
+                edge.setR(r);
+                break;
+            case G:
+                float g = Float.valueOf(value.toString());
+                if (g < 0.0 || g > 1.0) {
+                    throw new IllegalArgumentException("Color parameter outside of expected range: " + g);
+                }
+                edge.setG(g);
+                break;
+            case B:
+                float b = Float.valueOf(value.toString());
+                if (b < 0.0 || b > 1.0) {
+                    throw new IllegalArgumentException("Color parameter outside of expected range: " + b);
+                }
+                edge.setB(b);
+                break;
+            case COLOR:
+                Integer color = (value!=null)?Integer.decode(value.toString()):0;
+                int i = color.intValue();
+                edge.setR(((i >> 16) & 0xFF)/255.f);
+                edge.setG(((i >> 8) & 0xFF)/255.f);
+                edge.setB((i & 0xFF)/255.f);
+                break;
+            case SIZE:
+                float size = Float.valueOf(value.toString());
+                edge.setWeight(size);
+                break;
+        }
+    }
+
+    private void applyFilter(FilterEvent filterEvent) {
+        
+        Filter filter = filterEvent.getFilter();
+        if (filter instanceof NodeFilter) {
+            NodeFilter nodeFilter = (NodeFilter)filter;
+            Map<String, Object> attributes = filterEvent.getAttributes();
+            for (Node node: graph.getNodes().toArray()) {
+                if (nodeFilter.evaluate(graph, node)) {
+
+                    switch (filterEvent.getEventType()) {
+                        case ADD:
+                            log("Unsupported FilterEvent of type ADD");
+                            break;
+                        case CHANGE:
+                            graph.writeLock();
```

Comment:
Would probably be better to wrap the entire method on the writeLock to avoid side effects